pandas/tseries/index.py

```diff
@@ -762,6 +762,8 @@ def astype(self, dtype):
             return self.asi8.copy()
         elif dtype == _NS_DTYPE and self.tz is not None:
             return self.tz_convert('UTC').tz_localize(None)
+        elif dtype == str:
+            return Index(self.format(), dtype=object)
```

Comment:
I think this impl loses ``Index.name`` attribute.  Pls add ``name=self.name``. 

You can test this by passing ``name`` kw both in ``date_range`` and ``Index``.